trie/triedb/pathdb/database.go

```diff
@@ -193,30 +202,42 @@ func New(diskdb ethdb.Database, config *Config) *Database {
 		}
 		db.freezer = freezer
 
-		diskLayerID := db.tree.bottom().stateID()
-		if diskLayerID == 0 {
-			// Reset the entire state histories in case the trie database is
-			// not initialized yet, as these state histories are not expected.
-			frozen, err := db.freezer.Ancients()
-			if err != nil {
-				log.Crit("Failed to retrieve head of state history", "err", err)
-			}
-			if frozen != 0 {
-				err := db.freezer.Reset()
+		if db.isRecoverDiffLayers {
```

Comment:
Semantically, this code snippet is intended to clean up old data. Is it reasonable to put the load process here?

Reply:
Fixed